### PR TITLE
[ARCTIC-397][Core]Uprade iceberg version to 0.13.2

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

*Uprade iceberg version to 0.13.2*

## Brief change log

  - *Uprade iceberg version to 0.13.2*
  - *Change some dependencies*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/enumerator/AbstractArcticEnumerator.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void close() throws IOException {
   }
 
   @Override
-  public void handleSplitRequest(int subtaskId, @Nullable String requesterHostname) {
+  public void handleSplitRequest(int subtaskId, @Nonnull String requesterHostname) {
```
Comment (zstraw) [inline flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/enumerator/AbstractArcticEnumerator.java:71]: This is opposite annotation. Original means field can be null, but this Nonnull means can not be null.

**Review by zstraw — COMMENTED** [review]
All Nullable->NonNull should be fix

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/enumerator/AbstractArcticEnumerator.java (hunk 1 above)
Comment (zhoujinsong) [reply flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/enumerator/AbstractArcticEnumerator.java:71]: yes, I will fix it

**Review by zstraw — APPROVED** [review]
LGTM

**Pull request merged**